File: AutomaticBrewingCoffeeKioskBE/ABC_Kiosk_BE/Domain/Enums/OrderStatus.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Enums
{
    public enum OrderStatus
    {
        [Display(Name = "Pending")] Pending = 0,

        /// <summary>
        /// The order items are actively being prepared by the kiosk.
        /// </summary>
        [Display(Name = "Preparing")] Preparing = 1,

        /// <summary>
        /// The order has been successfully fulfilled
        /// </summary>
        [Display(Name = "Completed")] Completed = 2,

        /// <summary>
        /// The order was cancelled before Preparing
        /// </summary>
        [Display(Name = "Cancelled")] Cancelled = 3,

        /// <summary>
        /// The order could not be completed due to an unrecoverable error after confirmation (e.g., machine malfunction during preparation).
        /// </summary>
        [Display(Name = "Failed")] Failed = 4
    }
}

File: AutomaticBrewingCoffeeApp/lib/app/data/enums/order_status.dart

enum OrderStatus {
  pending,
  preparing,
  completed,
  cancelled,
  failed,
  //for all
  none
}

extension OrderStatusExtension on OrderStatus {
  String get value {
    switch (this) {
      case OrderStatus.pending:
        return 'Pending';
      case OrderStatus.preparing:
        return 'Preparing';
      case OrderStatus.completed:
        return 'Completed';
      case OrderStatus.cancelled:
        return 'Cancelled';
      case OrderStatus.failed:
        return 'Failed';
      case OrderStatus.none:
        return 'All';
    }
  }

  String get vnValue {
    switch (this) {
      case OrderStatus.pending:
        return 'Đang đợi';
      case OrderStatus.preparing:
        return 'Đang chuẩn bị';
      case OrderStatus.completed:
        return 'Hoàn thành';
      case OrderStatus.cancelled:
        return 'Hủy';
      case OrderStatus.failed:
        return 'Thất bại';
      case OrderStatus.none:
        return 'Tất cả';
    }
  }

  static List<String> get allValues =>
      OrderStatus.values.map((e) => e.value).toList();

  static OrderStatus fromValueStr(String value) {
    return OrderStatus.values.firstWhere((e) => e.value == value);
  }
}
